--- a/m4/x_ac_lwgrp.m4	2020-09-28 17:53:31.213286269 +0000
+++ b/m4/x_ac_lwgrp.m4	2020-09-28 17:53:54.089963629 +0000
@@ -47,10 +47,10 @@
     if test "$with_lwgrp" = check || \
        test "x$_x_ac_lwgrp_dirs" = xyes || \
        test "x$_x_ac_lwgrp_dirs" = "x" ; then
-#      AC_CHECK_LIB([lwgrp], [GCS_Comm_split])
+       AC_CHECK_LIB([lwgrp], [lwgrp_comm_free])
 
       # if we found it, set the build flags
-      if test "$ac_cv_lib_lwgrp_GCS_Comm_split" = yes; then
+      if test "$ac_cv_lib_lwgrp_lwgrp_comm_free" = yes; then
         found=yes
         LWGRP_CFLAGS=""
         LWGRP_LDFLAGS=""
--- a/configure	2017-11-05 01:52:57.000000000 +0000
+++ /bconfigure	2020-09-28 17:51:59.829595663 +0000
@@ -4945,10 +4945,54 @@
     if test "$with_lwgrp" = check || \
        test "x$_x_ac_lwgrp_dirs" = xyes || \
        test "x$_x_ac_lwgrp_dirs" = "x" ; then
-#      AC_CHECK_LIB([lwgrp], [GCS_Comm_split])
+       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for lwgrp_comm_free in -llwgrp" >&5
+$as_echo_n "checking for lwgrp_comm_free in -llwgrp... " >&6; }
+if ${ac_cv_lib_lwgrp_lwgrp_comm_free+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-llwgrp  $LIBS"
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char lwgrp_comm_free ();
+int
+main ()
+{
+return lwgrp_comm_free ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_lib_lwgrp_lwgrp_comm_free=yes
+else
+  ac_cv_lib_lwgrp_lwgrp_comm_free=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_lwgrp_lwgrp_comm_free" >&5
+$as_echo "$ac_cv_lib_lwgrp_lwgrp_comm_free" >&6; }
+if test "x$ac_cv_lib_lwgrp_lwgrp_comm_free" = xyes; then :
+  cat >>confdefs.h <<_ACEOF
+#define HAVE_LIBLWGRP 1
+_ACEOF
+
+  LIBS="-llwgrp $LIBS"
+
+fi
+
 
       # if we found it, set the build flags
-      if test "$ac_cv_lib_lwgrp_GCS_Comm_split" = yes; then
+      if test "$ac_cv_lib_lwgrp_lwgrp_comm_free" = yes; then
         found=yes
         LWGRP_CFLAGS=""
         LWGRP_LDFLAGS=""
